dask/array/tests/test_array_core.py

```diff
@@ -747,12 +768,165 @@ def test_where_bool_optimization():
         assert w1 is ex_w1
 
 
-def test_where_has_informative_error():
-    x = da.ones(5, chunks=3)
-    try:
-        da.where(x > 0)
-    except Exception as e:
-        assert 'dask' in str(e)
+def test_where_nonzero():
+    for shape, chunks in [(0, ()), ((0, 0), (0, 0)), ((15, 16), (4, 5))]:
+        x = np.random.randint(10, size=shape)
+        d = from_array(x, chunks=chunks)
+
+        x_w = np.where(x)
+        d_w = where(d)
+
+        assert isinstance(d_w, type(x_w))
+        assert len(d_w) == len(x_w)
+
+        for i in range(len(x_w)):
+            assert_eq(d_w[i], x_w[i])
+
+
+def test_where_incorrect_args():
+    a = da.ones(5, chunks=3)
+
+    for kwd in ["x", "y"]:
+        kwargs = {kwd: a}
+        try:
+            da.where(a > 0, **kwargs)
+        except ValueError as e:
+            assert 'either both or neither of x and y should be given' in str(e)
+
+
+def test_count_nonzero():
+    for shape, chunks in [(0, ()), ((0, 0), (0, 0)), ((15, 16), (4, 5))]:
+        x = np.random.randint(10, size=shape)
+        d = from_array(x, chunks=chunks)
+
+        x_c = np.count_nonzero(x)
+        d_c = da.count_nonzero(d)
+
+        if d_c.shape == tuple():
+            assert x_c == d_c.compute()
+        else:
+            assert_eq(x_c, d_c)
+
+
+@pytest.mark.skipif(LooseVersion(np.__version__) < '1.12.0',
+                    reason="NumPy's count_nonzero doesn't yet support axis")
+@pytest.mark.parametrize('axis', [None, 0, (1,), (0, 1)])
+def test_count_nonzero_axis(axis):
+    if (LooseVersion(np.__version__) < LooseVersion("1.12.0")):
+        pytest.skip(
+            "NumPy %s doesn't support multiple axes with `roll`."
+            " Need NumPy 1.12.0 or greater." % np.__version__
+        )
```

Comment:
Haha. Yeah, I think this is some detritus when I was playing with how to formulate these. Will clean it up.